rust/datafusion/src/logical_plan/mod.rs

```diff
@@ -755,8 +756,72 @@ impl fmt::Debug for Expr {
     }
 }
 
-/// The LogicalPlan represents different types of relations (such as Projection,
-/// Filter, etc) and can be created by the SQL query planner and the DataFrame API.
+/// This defines the interface for `LogicalPlan` nodes that can be
+/// used to extend DataFusion with custom relational operators.
+///
+/// See the example in
+/// [user_defined_plan.rs](../../tests/user_defined_plan.rs) for an
+/// example of how to use this extenison API
+pub trait ExtensionPlanNode: Debug {
```

Comment:
Another idea: `UserDefinedNode` (UDN), to be aligned with user defined function (UDF) that most people are familiar with and already know what it means.

Reply:
I like the idea of standardizing on `UserDefined*` for extension points. Everything we do here for logical plans, we will end up doing for physical plans too, I expect. Given that, we should add `Logical` to the name, so `UserDefinedLogicalNode` perhaps?

Reply:
I like `UserDefinedLogicalNode` -- I will make that change.